src/Shared/NativeMethodsShared.cs

```diff
@@ -299,16 +299,22 @@ public MemoryStatus()
         [StructLayout(LayoutKind.Sequential)]
         private struct PROCESS_BASIC_INFORMATION
         {
-            public IntPtr ExitStatus;
+            public uint ExitStatus;
             public IntPtr PebBaseAddress;
-            public IntPtr AffinityMask;
-            public IntPtr BasePriority;
-            public IntPtr UniqueProcessId;
-            public IntPtr InheritedFromUniqueProcessId;
+            public UIntPtr AffinityMask;
+            public int BasePriority;
+            public UIntPtr UniqueProcessId;
+            public UIntPtr InheritedFromUniqueProcessId;
 
-            public int Size
+            public uint Size
             {
-                get { return (6 * IntPtr.Size); }
+                get {
+#if (CLR2COMPATIBILITY)
+                    return (uint)Marshal.SizeOf(typeof(PROCESS_BASIC_INFORMATION));
```

Comment:
Can you make the method unsafe and use `sizeof(PROCESS_BASIC_INFORMATION)`? `Marshal.SizeOf` is very expensive.